Board.java

```diff
@@ -9,12 +9,20 @@ public Board(String state) {
         dimension = state.length() / (int) Math.sqrt(state.length());
     }
 
+    public int boardSize() {
+        return dimension * dimension;
+    }
+
     public String getState() {
         return this.state;
     }
 
      public Boolean availablePosition(int position) {
-         return position >= 0 && position <= dimension * dimension && state.charAt(position) == '-';
+         return validPositionOnBoard(position) && state.charAt(position) == '-';
```

Comment:
'-' could be an enum value: Empty ? along with 'x' and 'o'?

This: _state.charAt(position) == '-'_ could be a method _isEmpty()_